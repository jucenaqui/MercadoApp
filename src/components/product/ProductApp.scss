.product-list {
    background-color: white;
    padding: 10px;
    display: flex;
    flex-direction: column;
    justify-content: center;
    width: 80%;
    margin: 20px auto;
}