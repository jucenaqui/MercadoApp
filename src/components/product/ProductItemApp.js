import React from 'react';

import './ProductItemApp.scss';

export const ProductItemApp = ( { product}) => {
    
    return (
        <>
      
            <div className="product-item">
                <img className="producto-item-img" src={product.thumbnail}  alt={product.title}/>
                <div className="product-item-description"> 
                <h3>$ {product.price}</h3>   
                <p>{product.title}</p>
                </div>
            </div>
        </>
    )
}
