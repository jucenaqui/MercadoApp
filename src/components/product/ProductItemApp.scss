.product-item {
    display: flex;
    flex-wrap: wrap;
    justify-content: flex-start;
    padding: 10px 20px;
    box-shadow: 0px 1px 5px #f3f2f2;

    img {
        margin: 10px;
        border-radius: 3px;
        width: 100px;
    }
}

.product-item-description {
    align-items: flex-start;
    box-sizing: border-box;
    display: flex;
    flex-direction: column;
    flex-wrap: nowrap;
    padding-top: 20px;

    p {
        margin: 0 !important;
        width: 90%;
    }

    h3 {
        margin: 10px 0;
    }
}

@media screen and (max-width: '768px') {
    .product-item {
        flex-direction: column;
        justify-content: center;
        align-items: center;

        img {
            width: 150px;
        }
    }

    .product-item-description {
        align-items: center;

        p {
            width: 100%;
            padding: 5px 20px;
        }
    }
}