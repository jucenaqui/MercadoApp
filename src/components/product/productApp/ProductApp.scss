.product-list {
    background-color: white;
    display: flex;
    padding: 10px;
    flex-direction: column;
    justify-content: center;
    margin: 20px auto;
    width: 80%;
}

.product-category {
    color: gray;
    font-size: 1.2rem;
    font-weight: 600;
    margin: 20px auto;
    padding: 10px;
    width: 80%;
}
