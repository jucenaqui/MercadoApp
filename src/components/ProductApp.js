import React from 'react';
import {ProductItemApp} from './ProductItemApp';

const ProductApp = ( { products } ) => {

    return (
     <>
        <ol>
            {
                products.map((product)=> (<ProductItemApp key={product.id} product={product} />))
            }
        </ol>
     </>
    )
}

export default ProductApp;