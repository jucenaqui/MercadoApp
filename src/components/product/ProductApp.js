import React from 'react';
import {ProductItemApp} from './ProductItemApp';

import './ProductApp.scss';

const ProductApp = ( { products } ) => {

    return (
     <>
        <ol className="product-list">
            {
                products.map((product)=> (<ProductItemApp key={product.id} product={product} />))
            }
        </ol>
     </>
    )
}

export default ProductApp;