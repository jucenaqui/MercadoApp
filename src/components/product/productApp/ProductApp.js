import React from 'react';
import {ProductItemApp} from '../producto-item/ProductItemApp';
import PropTypes from 'prop-types';

import './ProductApp.scss';

const ProductApp = ( { products, categories } ) => {

    return (
     <>
        <p className="product-category"> 
        { 
            categories.map(item => item.name).join(' > ')
        } 
        </p>
        <ol className="product-list animate__animated animate__fadeIn">
            {
                products.map((product)=> (<ProductItemApp key={product.id} product={product} />))
            }
        </ol>
     </>
    )
}


ProductApp.propTypes = {
    products: PropTypes.array.isRequired,
    categories: PropTypes.array.isRequired
}


export default ProductApp;