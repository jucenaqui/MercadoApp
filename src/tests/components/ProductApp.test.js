import { shallow } from "enzyme";
import ProductApp from "../../components/product/productApp/ProductApp";
import { categoriesMuck, productsMuck } from "../mocks/product";

describe('Test to <ProductApp />', () => {
    
    const product = productsMuck;
    const categories = categoriesMuck;
    const wrapper = shallow(<ProductApp products={ productsMuck } categories={ categories } />);

    test('should show the component succesfull', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('should have a <p> with className product-category', () => { 
        const className = 'product-category';
        const p = wrapper.find('p');
        expect(p.prop('className').trim()).toBe(className);
    });

    test('should have a <p> with categories type string', () => { 
        const catgories = 'Electrodomésticos > Refrigeración > Refrigeradores';
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(catgories);
    });

    test('should have a <p> with className animate__fadeIn', () => { 
        const classAnimate = 'animate__fadeIn';
        const p = wrapper.find('ol');
        const className = p.prop('className');
        expect(className.includes(classAnimate)).toBe(true);
    });

})
