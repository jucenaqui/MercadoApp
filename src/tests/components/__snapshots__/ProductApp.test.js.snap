// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test to <ProductApp /> should show the component succesfull 1`] = `
<Fragment>
  <p
    className="product-category"
  >
    Electrodomésticos &gt; Refrigeración &gt; Refrigeradores
  </p>
  <ol
    className="product-list animate__animated animate__fadeIn"
  >
    <ProductItemApp
      key="MLA856481792"
      product={
        Object {
          "condition": "new",
          "description": "la description",
          "free_shipping": false,
          "id": "MLA856481792",
          "picture": "http://http2.mlstatic.com/D_965028-MLA32542008594_102019-I.jpg",
          "price": Object {
            "amount": 47200,
            "currency": "ARS",
            "decimals": 0,
          },
          "title": "Heladera",
        }
      }
    />
    <ProductItemApp
      key="MLA856481792"
      product={
        Object {
          "condition": "old",
          "description": "la description 2",
          "free_shipping": false,
          "id": "MLA856481792",
          "picture": "http://http2.mlstatic.com/D_965028-MLA32542008594_102019-I.jpg",
          "price": Object {
            "amount": 47300,
            "currency": "ARS",
            "decimals": 0,
          },
          "title": "Heladera2",
        }
      }
    />
  </ol>
</Fragment>
`;
